integrationm/test/com/cultofbits/customizations/audit/AuditUtilsTest.groovy

```diff
@@ -0,0 +1,239 @@
+        def newInstanceMsg = RecordmMsgBuilder.aMessage("cob-admin", definition, "update").newField(definition.getFirstField("Text"), "text")
+                .build()
+        RecordmMsg.Field f = Mock(RecordmMsg.Field.class)
+        f.changed() >> false
+        newInstanceMsg.field("") >> f
+
+        def auditFields = AuditInstanceUpdater.getAuditFields(null, newInstanceMsg, definition)
+
+        expect:
+        definition.rootFields.size() == 6
+        auditFields.size() == 4
+        auditFields[0].name == "Creator"
+        auditFields[0].op == "creator"
+        auditFields[0].args == "username"
+
+
+        auditFields[1].name == "TimeCreation"
+        auditFields[1].op == "updater"
+        auditFields[1].args == "time"
+
+
+        auditFields[2].name == "TimeCreation2"
+        auditFields[2].op == "updater"
+        auditFields[2].args == "uri"
+
+
+        auditFields[3].name == "TimeCreation3"
+        auditFields[3].op == "updater"
+        auditFields[3].args == "username"
+    }
+
+    def "when none of the nonRef fields change and ignore audit is true"() {
+        def definitionBuilder = getDefinition()
+
+        def definition = definitionBuilder.addFieldDefinitions(aFieldDefinition()
+                .id(14)
+                .name("TimeCreation2")
+                .description("\$audit.updater.uri([ignoreRefs:false,a:1])")
+                .setConfigurationExtensions("\$audit.updater.time", "{\"args\":[\"[ignoreRefs:false,a:1]\"]}}}")
+                .build())
+                .addFieldDefinitions(aFieldDefinition()
+                        .id(16)
+                        .name("TimeCreation3")
+                        .description("\$audit.updater.username([ignoreRefs:true,a:1])")
+                        .setConfigurationExtensions("\$audit.updater.time", "{\"args\":[\"[ignoreRefs:true,a:1]\"]}}}")
+                        .build())
+                .build();
+
+        RecordmMsg newInstanceMsg = RecordmMsgBuilder.aMessage("cob-admin", definition, "update").build()
+        RecordmMsg.Field f = Stub()
+        f.changed() >> false
+        newInstanceMsg.field("") >> f
+
+        def auditFields = AuditInstanceUpdater.getAuditFields(null, newInstanceMsg, definition)
+
+        expect:
+        auditFields.size() == 3
+    }
+
+    def "when at least one nonRef field changes and ignore audit is true"() {
+        def definitionBuilder = getDefinition()
+
+        def definition = definitionBuilder.addFieldDefinitions(aFieldDefinition()
+                .id(14)
+                .name("TimeCreation2")
+                .description("\$audit.updater.uri([ignoreRefs:false,a:1])")
+                .build())
+                .addFieldDefinitions(aFieldDefinition()
+                        .id(16)
+                        .name("TimeCreation3")
+                        .description("\$audit.updater.username([ignoreRefs:true,a:1])")
+                        .build())
+                .build();
```

Comment:
You're doing this a lot of times. Maybe you could provide this in the 
`getDefinition(FieldDefinitionBuilders.... extraFields)`